language/move-vm/runtime/src/unit_tests/arguments.rs

```diff
@@ -0,0 +1,119 @@
+    },
+};
+
+fn make_script_with_sig(signature: Signature) -> Vec<u8> {
+    let mut blob = vec![];
+    CompiledScriptMut {
+        module_handles: vec![],
+        struct_handles: vec![],
+        function_handles: vec![],
+
+        function_instantiations: vec![],
+
+        signatures: vec![Signature(vec![]), signature],
+
+        identifiers: vec![],
+        address_identifiers: vec![],
+        constant_pool: vec![],
+
+        type_parameters: vec![],
+        parameters: SignatureIndex(1),
+        code: CodeUnit {
+            locals: SignatureIndex(0),
+            code: vec![Bytecode::Ret],
+        },
+    }
+    .serialize(&mut blob)
+    .expect("script must serialize");
+    blob
+}
+
+struct RemoteStore {}
+
+impl RemoteCache for RemoteStore {
+    fn get_module(&self, _module_id: &ModuleId) -> VMResult<Option<Vec<u8>>> {
+        Ok(None)
+    }
+
+    fn get_resource(
+        &self,
+        _address: &AccountAddress,
+        _tag: &TypeTag,
+    ) -> PartialVMResult<Option<Vec<u8>>> {
+        Ok(None)
+    }
+}
+
+fn call_script(script: Vec<u8>, args: Vec<Value>) -> VMResult<()> {
+    let move_vm = MoveVM::new();
+    let remote_view = RemoteStore {};
+    let mut session = move_vm.new_session(&remote_view);
+    let cost_table = zero_cost_schedule();
+    let mut cost_strategy = CostStrategy::system(&cost_table, GasUnits::new(0));
+    session.execute_script(
+        script,
+        vec![],
+        args,
+        AccountAddress::random(),
+        &mut cost_strategy,
+    )
+}
+
+#[test]
+fn simple_args() {
+    // no args provided
+    let script = make_script_with_sig(Signature(vec![SignatureToken::U64]));
+    call_script(script, vec![]).expect_err("missing arguments");
+
+    // too many args
+    let script = make_script_with_sig(Signature(vec![SignatureToken::Bool]));
+    call_script(script, vec![Value::bool(true), Value::u8(0)]).expect_err("too many arguments");
+
+    // right args
+    let script = make_script_with_sig(Signature(vec![SignatureToken::U128]));
+    call_script(script, vec![Value::u128(0)]).expect("good u128 failed");
+
+    // vector of address
+    let script = make_script_with_sig(Signature(vec![SignatureToken::Vector(Box::new(
+        SignatureToken::Address,
+    ))]));
+    call_script(script, vec![Value::vector_address(vec![])]).expect("good vector<address> failed");
+    let script = make_script_with_sig(Signature(vec![SignatureToken::Vector(Box::new(
+        SignatureToken::Address,
+    ))]));
+    call_script(
+        script,
+        vec![Value::vector_address(vec![AccountAddress::random()])],
+    )
+    .expect("good vector<address> failed");
+
+    // bad vector of address
+    let script = make_script_with_sig(Signature(vec![SignatureToken::Vector(Box::new(
+        SignatureToken::Address,
+    ))]));
+    call_script(script, vec![Value::vector_bool(vec![true])]).expect_err("bad vector<address>");
+
+    // bad vector of vector
```

Comment:
Is "good vector of vector" one of the tests coming later?

Reply:
yeah there is more test coming as I said, but I need to be able to build those and the runtime is not that nice towards that, so I stopped.
To be clear, thanks for the approval but this was not going in until tomorrow when I get time to add all the tests.
Having said so, the verifier mistake was there so that is great!